### ISSUE TITLE
[Discussion] Public cloud share content providers

### ISSUE THREAD

**Issue by nuest** [jupyter/repo2docker#805]
### Proposed change

Add content providers to turn a public cloud share link into a Binder, e.g. an ownCloud share (but also Dropbox, Google Drive). These services are getting widespread in educational contexts and could be a way to make Binder accessible to new user groups.

### Alternative options

This might be deemed out of scope, so I'm mainly leaving this here for others to share their opinions, and to make sure others interested in this feature can pinch in.

### Who would use this feature?

Users with computational workflows outside of git and data repositories, but in cloud storage.

### How much effort will adding it take?

Really depends on the API of the respective cloud provider. In case of ownCloud with potentially many instances, detecting that a link is an ownCloud (or NextCloud) share might be tricky.

### Who can do this work?

Familiarity with the content provider infrastructure and with web APIs.

**Comment by manics** [comment]
Do you think it'd be easy to for repo2docker to support content providers as plugins? Then anyone could publish their own `repo2docker-xxx-provider` on pypi, and if it sees widespread use it could be merged into the main repo?

**Comment by betatim** [comment]
We discussed using entrypoints as a way to allow people to write content providers (and buildpacks) without having to add them to repo2docker directly in #488. I think the lack of opposition means people support the idea (but no one has gotten around to building it yet).

**Comment by nuest** [comment]
@manics I agree using some kind of plug-in mechanism is sensible with the growing number of content providers. How would you determine "widespread use"? Do you mean the different plug-ins would be installed on MyBinder.org and moved into the main repo if used often?

**Comment by manics** [comment]
@nuest Good question! I was thinking about long term maintenance for the different providers. Once a provider is added to mybinder.org is there an assumption it'll be supported forever? If so what are the minimum requirements for a new provider to be added? If other BinderHubs customise their installations with additional providers it might be possible to get stats on how popular they are, and the provider could then "graduate" to mybinder.
